### PR TITLE
ALTAPPS-886: [Android, Shared] update UI in mobile app for limits

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[#ALTAPPS-886](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-886)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Remove `stepsLimitTotal`, `stepsLimitLeft` from `ProblemsLimitFeature.ViewState.Widget`
- Add progress to `ProblemsLimitFeature.ViewState.Widget`
- Update texts & date formatting
- Update Android UI

### CODE REVIEW

File: androidHyperskillApp/src/main/res/layout/layout_problems_limit.xml [hunk 1]
```diff
@@ -23,28 +30,32 @@
...
             android:id="@+id/problemsLimitUpdatedIn"
             android:layout_width="wrap_content"
-            android:layout_height="match_parent"
+            android:layout_height="wrap_content"
             android:textColor="@color/color_on_surface_alpha_38"
             android:textSize="12sp"
             android:gravity="bottom"
-            tools:text="Update in 2 h."/>
+            android:layout_gravity="center_vertical"
+            tools:text="Reset in 2 h. Reset in 2 h. Reset in 2 h. Reset in 2 h."/>
```
Comment (vladkash) [inline androidHyperskillApp/src/main/res/layout/layout_problems_limit.xml:44]: it seems better to leave "Reset in 2 h." Like we have it in design

File: shared/src/commonMain/resources/MR/base/plurals.xml [hunk 2]
```diff
@@ -12,6 +12,14 @@
...
         <item quantity="other">%d hours</item>
     </plural>
+    <plural name="minutes_short">
+        <item quantity="one">%d m.</item>
+        <item quantity="other">%d m.</item>
+    </plural>
+    <plural name="hours_short">
+        <item quantity="one">%d h.</item>
+        <item quantity="other">%d h.</item>
+    </plural>
```
Comment (vladkash) [inline shared/src/commonMain/resources/MR/base/plurals.xml:22]: If we have the same vales for `one` and `other`. Maybe we can use only `other` item or just move that resource to `strings.xml`

**Review by vladkash — APPROVED** [review]

File: androidHyperskillApp/src/main/res/layout/layout_problems_limit.xml (hunk 1 above)
Comment (XanderZhu) [reply androidHyperskillApp/src/main/res/layout/layout_problems_limit.xml:44]: @vladkash 
A long string is used here to test the behavior in the case of a small device when this text view doesn't have enough width.

File: shared/src/commonMain/resources/MR/base/plurals.xml (hunk 2 above)
Comment (XanderZhu) [reply shared/src/commonMain/resources/MR/base/plurals.xml:22]: @vladkash 
You are right, in this case we case use just a string not plurals

**Pull request merged**